web-server-lib/web-server/http/request.rkt

```diff
@@ -26,6 +26,9 @@
 ;; Happy hacking!
 ;;
 
+(define nonnegative-length/c
+  (or/c exact-nonnegative-integer? +inf.0))
```

Comment:
Bikeshedding about names: if we use this for more things as we were discussing in <https://github.com/racket/web-server/pull/77>, it would apply to timeouts and counts (e.g. `#:max-request-files`) as well as "length"s per se. I don't have a great idea for a name, but maybe something like `nonnegative-limit/c`?